tools/codegen/gen_backend_stubs.py

```diff
@@ -171,6 +171,27 @@ def create_decl(f: NativeFunction) -> str:
 """
     assert missing_kernels_err_msg == "", missing_kernels_err_msg
 
+    # Additionally, provide a nice error if a user tries to register a composite op with no derivative formula to a backend key.
+    # Otherwise, autograd will silently stop working for their op.
+    for func in native_functions:
+        if backend_indices[backend_key].has_kernel(func) and func.has_composite_implicit_autograd_kernel:
+            # At this point, we know that the vendor provided a kernel for an op that also has a CompositeImplicitAutograd kernel.
+            # What we really want to error on, though, is if that op doesn't have a derivative formula registered in-tree.
+            # We can guess that based on whether or not the op has ANY in-tree non-composite kernels.
+            # Technically this has a small chance of false negatives, but it's easier than re-parsing derivatives.yaml here.
+            idxs = [idx for idx in backend_indices.values() if not idx.external and idx.has_kernel(func)]
+            has_non_composite_in_tree_backend = any(
+                idx.has_kernel(func) for idx in backend_indices.values()
+                if not idx.external and idx.dispatch_key != DispatchKey.CompositeImplicitAutograd
+                and idx.dispatch_key != DispatchKey.CompositeExplicitAutograd)
+            assert has_non_composite_in_tree_backend, \
+                f"\nFound an entry for '{func.func.name}' in the yaml. " \
```

Comment:
in the yaml for {backend_key}. " \

we may also print the dispatch key since one backend might have several dispatch keys.